flake.nix

```diff
@@ -1,75 +1,117 @@
 {
-  description =
-    "ripgrep, but also search in PDFs, E-Books, Office documents, zip, tar.gz, etc.";
+  description = "ripgrep, but also search in PDFs, E-Books, Office documents, zip, tar.gz, etc.";
 
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs";
-
     crane = {
       url = "github:ipetkov/crane";
       inputs.nixpkgs.follows = "nixpkgs";
     };
-
-    flake-utils.url = "github:numtide/flake-utils";
-
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    flake-root.url = "github:srid/flake-root";
     rust-overlay = {
       url = "github:oxalica/rust-overlay";
-      inputs = {
-        nixpkgs.follows = "nixpkgs";
-        flake-utils.follows = "flake-utils";
-      };
+      inputs.nixpkgs.follows = "nixpkgs";
     };
-
     advisory-db = {
       url = "github:rustsec/advisory-db";
       flake = false;
     };
-
     pre-commit-hooks = {
       url = "github:cachix/pre-commit-hooks.nix";
-      inputs = {
-        nixpkgs.follows = "nixpkgs";
-        flake-utils.follows = "flake-utils";
-      };
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    treefmt-nix = {
+      url = "github:numtide/treefmt-nix";
+      inputs.nixpkgs.follows = "nixpkgs";
     };
   };
 
-  outputs = { self, nixpkgs, crane, flake-utils, rust-overlay, advisory-db
-    , pre-commit-hooks }:
-    flake-utils.lib.eachDefaultSystem (system:
-      let
-        pkgs = import nixpkgs {
+  outputs = inputs:
+    inputs.flake-parts.lib.mkFlake {inherit inputs;} {
+      imports = [
+        inputs.flake-root.flakeModule
+        inputs.pre-commit-hooks.flakeModule
+        inputs.treefmt-nix.flakeModule
+      ];
+      systems = ["x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin"];
+      perSystem = {
+        config,
+        self',
+        inputs',
+        system,
+        ...
+      }: let
+        pkgs = import inputs.nixpkgs {
           inherit system;
-          overlays = [ (import rust-overlay) ];
+          overlays = [(import inputs.rust-overlay)];
         };
 
-        craneLib = crane.lib.${system};
+        craneLib = inputs.crane.lib."${system}";
         src = pkgs.lib.cleanSourceWith {
           src = craneLib.path ./.;
           filter = pkgs.lib.cleanSourceFilter;
         };
 
-        buildInputs = with pkgs;
-          [ ffmpeg imagemagick pandoc poppler_utils ripgrep tesseract ]
-          ++ pkgs.lib.optionals pkgs.stdenv.isDarwin [
-            # Additional darwin specific inputs can be set here
-            pkgs.libiconv
-          ];
+        nativeBuildInputs = pkgs.lib.optionals pkgs.stdenv.isDarwin [
+          # Additional darwin specific inputs can be set here
+          pkgs.libiconv
+        ];
+
+        runtimeInputs = with pkgs; [
+          ffmpeg
+          imagemagick
+          pandoc
+          poppler_utils
+          ripgrep
+          tesseract
+          zip
+        ];
```

Comment:
tesseract and imagemagick are not used anymore

Reply:
Will remove them. Thanks!

Reply:
Removed in 4e8dd8f7a0a7d9091e088bf89c61ce6e24707864.